src/geometry/adjoints.jl

```diff
@@ -49,3 +49,50 @@ function ChainRulesCore.rrule(::Type{SVector{N, T}}, x...) where {N, T}
     end
     return y, svector_const_pb
 end
+
+function ChainRulesCore.rrule(::typeof(ImageTransformations._getindex), A::AbstractExtrapolation, x)
+    y = A(Tuple(x)...)
+    function _getindex_pb(Δy)
+        # Δy :: NamedTuple{(:r, :g, :b)} or something similar
+        Δy = eltype(A)(Δy...)
+        gr = Interpolations.gradient(A, Tuple(x)...)
+        n = nfields(Δy) > 0 ? nfields(Δy) : 1
+        return NoTangent(), NoTangent(), Δy .⋅ gr .⋅ n
```

Comment:
Urgh..., what is `⋅` here? Maybe write in a more obvious way?

Reply:
Haha it is this:
```julia
[∂J/∂i ∂J/∂j] * [∂i/∂W, ∂j/∂W]
```
Since both `[∂J/∂i ∂J/∂j]` and `[∂i/∂W, ∂j/∂W]` can be in the form of `Colorant`s, I use the `\cdot[tab]` from ColorVectorSpace defined for `<: Colorants`. And since the CVC `⋅`(`dot`) [normalizes the answer](https://github.com/JuliaGraphics/ColorVectorSpace.jl#multiplication), I use the `n` variable to denormalize it. It looked clean, and therefore I kept it that way :p